lib/pure/json.nim

```diff
@@ -1103,7 +1103,7 @@ when defined(nimFixedForwardGeneric):
       jsonPath.add '['
       jsonPath.addInt i
       jsonPath.add ']'
-      initFromJson(dst[i], jsonNode[i], jsonPath)
+      initFromJson(dst[i.S], jsonNode[i], jsonPath) # `.S` for enum indexed arrays
```

Comment:
did you rebase against devel? if you did, this diff shouldn't show up given that  https://github.com/nim-lang/Nim/pull/16807 was merged